perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathWatcherFactory.java

```diff
@@ -0,0 +1,107 @@
+package name.mitterdorfer.perlock;
+
+import name.mitterdorfer.perlock.impl.util.Preconditions;
+
+import java.nio.file.Path;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * Factory to create {@link PathWatcher}s that only deal with a single file and
+ * ignore change to other files within the directory
+ */
+public class SinglePathWatcherFactory {
```

Comment:
I think a separate factory is a fairly large change. I understand that you did not want to change the existing API but I think overall it would be better to merge this class with `PathWatcherFactory`.

What about a new public method `createSingleFileWatcher()` in `PathWatcherFactory`? It should also have a precondition check that a `Path` representing a file is provided.